src/coreclr/src/jit/hwintrinsiccodegenarm64.cpp

```diff
@@ -156,6 +156,11 @@ void CodeGen::genHWIntrinsic(GenTreeHWIntrinsic* node)
     {
         emitSize = EA_SIZE(node->gtSIMDSize);
         opt      = genGetSimdInsOpt(emitSize, intrin.baseType);
+
+        if ((opt == INS_OPTS_1D) && (intrin.category == HW_Category_SimpleSIMD))
+        {
+            opt = INS_OPTS_NONE;
```

Comment:
Why do we need to strip the opt here? Is knowing it is a 1D SimpleSIMD no longer important?

Reply:
As far I understood, most of the SimpleSIMD doesn't support 1D arrangement (usually the corresponding encoding marked as reserved) - except for maybe LD1, ST1 and other memory load/store instructions. Since we want to use a scalar instruction for such intrinsics I am removing the opt.

Reply:
Alternatively, we could mark all the "scalar" intrinsics as ScalarSIMD in hwintrinsiclistarm64.h. It is not a case right now - `BitwiseSelect` doesn't have a `BitwiseSelectScalar` version and all the intrinsics are SimpleSIMD.

However, this would also imply that during instruction lookup by intrinsic name we would need also take into account the full vector SIMD type (not just the element type) since we would need to distinguish between `Vector64<double>` and `Vector128<double>`.

Reply:
I'm not sure I got the last part, for x86 we use `HW_Category_SIMDScalar` to mark them and for ARM, all "scalar" intrinsics would be Vector64, no?

Reply:
What I am trying to say that if we want to have some of the `BitwiseSelect` variants (e.g. operating on `Vector64<double>` but not on `Vector64<int>`) marked as `HW_Category_SIMDScalar` and all others marked as `HW_Category_SimpleSIMD` we would need some way to define this in in hwintrinsiclistarm64.h. 

I don't think you can do this in the current implementation.

Reply:
Yes, but I'm also not aware of anything that would fall into this scenario; plus the naming convention has generally been `X` and `XScalar` to differentiate (as you couldn't distinguish something that oeprated on `Vector64<float>` as a scalar vs `Vector64<float>` as a vector by overload alone).

Reply:
I think we can continue our discussion - I don't want to block the PR on this though.